From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: pmos-laurel-sprout <dev@pmos.laurel.invalid>
Date: 2026-08-03
Subject: [PATCH] arm64: dts: qcom: sm6125: add Adreno 610 GPU, GMU wrapper, GPUCC y SMMU

Porta los nodos GPU de sm61x5-mainline (barni2000/7.0-develop,
base 7.0.8) a mainline v7.1. La GPU (gpu@5900000, compatible
qcom,adreno-610.0) es downstream-only; mainline v7.1 tiene el driver
msm/adreno con soporte a610 y gmu-wrapper (a6xx_gpu.c) pero no el DT.

Adaptaciones respecto a la fuente:
- regs convertidos de 2-cell a 1-cell (v7.1 soc@0 usa #address-cells = <1>)
- power-domains = <&rpmpd RPMPD_VDDCX> (v7.1 incluye qcom-rpmpd.h genérico;
  el per-SoC qcom-sm6125-rpmpd.h NO existe en v7.1)
- include dt-bindings/clock/qcom,sm6125-gpucc.h (el header existe en v7.1;
  qcom,gpucc-sm6125.h NO existe)

Añade: gpu@5900000 (opp-table), gmu_wrapper@596a000, gpucc@5990000,
adreno_smmu@59a0000 (iommu@59a0000), todos status="disabled" (el enable
va en el DTS de placa). GPU = fase 2; Plasma funciona con llvmpipe.


Signed-off-by: pmos-laurel-sprout <dev@pmos.laurel.invalid>
---
 arch/arm64/boot/dts/qcom/sm6125.dtsi | 125 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 125 insertions(+), 0 deletion(-)

diff --git a/arch/arm64/boot/dts/qcom/sm6125.dtsi b/arch/arm64/boot/dts/qcom/sm6125.dtsi
--- sm6125-v71.dtsi	2026-08-02 23:24:34.263589229 -0500
+++ sm6125-v71-new.dtsi	2026-08-02 23:39:10.747714583 -0500
@@ -5,8 +5,9 @@
 
 #include <dt-bindings/clock/qcom,dispcc-sm6125.h>
 #include <dt-bindings/clock/qcom,dsi-phy-28nm.h>
 #include <dt-bindings/clock/qcom,gcc-sm6125.h>
+#include <dt-bindings/clock/qcom,sm6125-gpucc.h>
 #include <dt-bindings/clock/qcom,rpmcc.h>
 #include <dt-bindings/dma/qcom-gpi.h>
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/interrupt-controller/arm-gic.h>
@@ -1246,8 +1247,132 @@
 			compatible = "qcom,rpm-stats";
 			reg = <0x04690000 0x10000>;
 		};
 
+		gpu: gpu@5900000 {
+			compatible = "qcom,adreno-610.0", "qcom,adreno";
+			reg = <0x05900000 0x40000>;
+			reg-names = "kgsl_3d0_reg_memory";
+
+			clocks = <&gpucc GPU_CC_GX_GFX3D_CLK>,
+				 <&gpucc GPU_CC_AHB_CLK>,
+				 <&gcc GCC_BIMC_GPU_AXI_CLK>,
+				 <&gcc GCC_GPU_MEMNOC_GFX_CLK>,
+				 <&gpucc GPU_CC_CX_GMU_CLK>,
+				 <&gpucc GPU_CC_CXO_CLK>;
+			clock-names = "core",
+				      "iface",
+				      "mem_iface",
+				      "alt_mem_iface",
+				      "gmu",
+				      "xo";
+
+			interrupts = <GIC_SPI 177 IRQ_TYPE_LEVEL_HIGH>;
+
+			iommus = <&adreno_smmu 0 1>;
+			operating-points-v2 = <&gpu_opp_table>;
+			power-domains = <&rpmpd RPMPD_VDDCX>;
+			qcom,gmu = <&gmu_wrapper>;
+
+			#cooling-cells = <2>;
+
+			status = "disabled";
+
+			zap-shader {
+				memory-region = <&gpu_mem>;
+			};
+
+			gpu_opp_table: opp-table {
+				compatible = "operating-points-v2";
+
+				opp-320000000 {
+					opp-hz = /bits/ 64 <320000000>;
+					required-opps = <&rpmpd_opp_low_svs>;
+				};
+
+				opp-465000000 {
+					opp-hz = /bits/ 64 <465000000>;
+					required-opps = <&rpmpd_opp_svs>;
+				};
+
+				opp-600000000 {
+					opp-hz = /bits/ 64 <600000000>;
+					required-opps = <&rpmpd_opp_svs_plus>;
+				};
+
+				opp-745000000 {
+					opp-hz = /bits/ 64 <745000000>;
+					required-opps = <&rpmpd_opp_nom>;
+				};
+
+				opp-820000000 {
+					opp-hz = /bits/ 64 <820000000>;
+					required-opps = <&rpmpd_opp_nom_plus>;
+				};
+
+				opp-900000000 {
+					opp-hz = /bits/ 64 <900000000>;
+					required-opps = <&rpmpd_opp_turbo>;
+				};
+
+				opp-950000000 {
+					opp-hz = /bits/ 64 <950000000>;
+					required-opps = <&rpmpd_opp_turbo_no_cpr>;
+				};
+			};
+		};
+
+		gmu_wrapper: gmu@596a000 {
+			compatible = "qcom,adreno-gmu-wrapper";
+			reg = <0x0596a000 0x30000>;
+			reg-names = "gmu";
+			power-domains = <&gpucc GPU_CX_GDSC>,
+					<&gpucc GPU_GX_GDSC>;
+			power-domain-names = "cx", "gx";
+
+			status = "disabled";
+		};
+
+		gpucc: clock-controller@5990000 {
+			compatible = "qcom,sm6125-gpucc";
+			reg = <0x05990000 0x9000>;
+			clocks = <&rpmcc RPM_SMD_XO_CLK_SRC>,
+				 <&gcc GCC_GPU_GPLL0_CLK_SRC>;
+			#clock-cells = <1>;
+			#power-domain-cells = <1>;
+
+			status = "disabled";
+		};
+
+		adreno_smmu: iommu@59a0000 {
+			compatible = "qcom,sm6125-smmu-500", "qcom,adreno-smmu",
+				     "qcom,smmu-500", "arm,mmu-500";
+			reg = <0x059a0000 0x10000>;
+			interrupts = <GIC_SPI 163 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 167 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 168 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 169 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 170 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 171 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 172 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 173 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 174 IRQ_TYPE_LEVEL_HIGH>;
+
+			clocks = <&gcc GCC_GPU_MEMNOC_GFX_CLK>,
+				 <&gpucc GPU_CC_HLOS1_VOTE_GPU_SMMU_CLK>,
+				 <&gcc GCC_GPU_SNOC_DVM_GFX_CLK>;
+			clock-names = "mem",
+				      "hlos",
+				      "iface";
+			power-domains = <&gpucc GPU_CX_GDSC>;
+
+			#global-interrupts = <1>;
+			#iommu-cells = <2>;
+
+			status = "disabled";
+		};
+
+
 		mdss: display-subsystem@5e00000 {
 			compatible = "qcom,sm6125-mdss";
 			reg = <0x05e00000 0x1000>;
 			reg-names = "mdss";

